hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -1530,13 +1559,18 @@ def draw_float(
             assert allow_nan or not math.isnan(forced)
             assert math.isnan(forced) or min_value <= forced <= max_value
 
-        return self.provider.draw_float(
-            min_value=min_value,
-            max_value=max_value,
-            allow_nan=allow_nan,
-            smallest_nonzero_magnitude=smallest_nonzero_magnitude,
-            forced=forced,
-        )
+        kwargs = {
+            "min_value": min_value,
+            "max_value": max_value,
+            "allow_nan": allow_nan,
+            "smallest_nonzero_magnitude": smallest_nonzero_magnitude,
+        }
+        value = self.provider.draw_float(**kwargs, forced=forced)
+        if observe:
```

Comment:
I wonder if this would cause problems when forcing values a la https://github.com/HypothesisWorks/hypothesis/pull/3806? If we realize the buffer of a third party provider, none of the examples would be present with this, which may cause e.g. bad shrinking.

Of course, we're doing away with the buffer eventually, and until then we can probably realize the buffer via forced and then re-run a vanilla ConjectureData on that buffer to obtain the examples.

Reply:
this basically omits all forced draws from the tree, right? I feel like there has to be a good reason this hasn't been done yet...

One thing is we lose the flakiness detection on forced values. I don't know how useful that is in detecting flakiness we would have missed otherwise in practice, but it does make me wary of changing this. 

We force pretty rarely anyway, right? I'm not too worried about performance here, but you know better than I.

Reply:
IIRC a forced node must always be the sole child of its parent, right?  Or else flaky, so the only thing we'd be giving up is flaky-detection (which is in fact nice to have).  Those test results aren't hugely encouraging, so lets keep the status quo - if we want to improve perf we can do some profiling later.